filament/src/materials/dofUtils.fs

```diff
@@ -0,0 +1,71 @@
+
+/*
+ * DoF blur
+ *
+ * This uses a lot of ideas from
+ *   "Bokeh depth of field in a single pass" by Dennis Gustafsson
+ *   (http://blog.tuxedolabs.com/2018/05/04/bokeh-depth-of-field-in-single-pass.html)
+ *
+ * needs:
+ *   materialParams_depth : depth texture
+ *   materialParams.cocParams : coc scale and bias
+ *   materialParams.resolution : screen resolution
+ */
+
+// Filter sample count, prefer odd values
+// (keep in sync with PostProcessManager.cpp:dof())
+const float SAMPLE_COUNT = 25.0;
+
+// A value > 1.0 allows a larger blur w/ dithering
+const float BLUR_SCALE = 1.0;
+
+#define unitvec(angle) vec2(cos(angle), sin(angle))
+
+float random(vec2 w) {
```

Comment:
Add a comment explaining this is interleaved gradient noise
Even better, use our function `interleavedGradientNoise()` from `dither.fs`.

Also `vec2 w` needs to be `highp`.